src/platform-aws.c

```diff
@@ -308,6 +310,33 @@ static int configure_ep_inorder(struct fid_ep *ep, int optname, const char* optn
 	return 0;
 }
 
+/*
+ * For the RDMA protocol, try to set max msg size on the current endpoint
+ * to the size of the max message we send with fi_send. This allows the EFA
+ * provider to enable the zero-copy path.
+ *
+ * Returns 0 on success or -error code on unexpected failure.
+ */
+static int configure_ep_max_msg_size(struct fid_ep *ep)
+{
+	int ret = 0;
+
+	size_t eager_max_size = (size_t)ofi_nccl_eager_max_size();
+	size_t optval = NCCL_OFI_MAX(NCCL_OFI_MAX(sizeof(nccl_net_ofi_rdma_ctrl_msg_t), eager_max_size),
+				     sizeof(nccl_ofi_rdma_connection_info_t));
+
+	ret = fi_setopt(&ep->fid, FI_OPT_ENDPOINT, FI_OPT_MAX_MSG_SIZE, &optval, sizeof(optval));
+	if (ret == -FI_EOPNOTSUPP || ret == -FI_ENOPROTOOPT) {
+		NCCL_OFI_INFO(NCCL_INIT, "Setting FI_OPT_MAX_MSG_SIZE not supported.");
```

Comment:
I guess you don't want to treat this as an error. In that case, I think you need to set `ret = 0` here to avoid propagating a failure to the caller.

Also, I think we'll need some sort of preprocessor wrapper around this code, so the plugin can compile with Libfabric versions that don't have `FI_OPT_MAX_MSG_SIZE`. E.g., see how we handle `FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES`.